src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1234,9 +1233,11 @@ private function startClass(string $class, string $baseClass, bool $hasProxyClas
 class $class extends $baseClass
 {
     protected \$parameters = [];
+    protected readonly \WeakReference \$ref;
```

Comment:
could we make it private ? AFAICT, the parent class never uses it directly.

Reply:
The standalone factories we generate need protected access.